pkgs/top-level/default.nix

```diff
@@ -41,45 +41,59 @@
 } @ args:
 
 let # Rename the function arguments
-  config0 = config;
   crossSystem0 = crossSystem;
 
 in let
   lib = import ../../lib;
 
-  # Allow both:
-  # { /* the config */ } and
-  # { pkgs, ... } : { /* the config */ }
-  config1 =
-    if lib.isFunction config0
-    then config0 { inherit pkgs; }
-    else config0;
-
   # From a minimum of `system` or `config` (actually a target triple, *not*
   # nixpkgs configuration), infer the other one and platform as needed.
   localSystem = lib.systems.elaborate (
     # Allow setting the platform in the config file. This take precedence over
     # the inferred platform, but not over an explicitly passed-in one.
-    builtins.intersectAttrs { platform = null; } config1
+    builtins.intersectAttrs { platform = null; } config
     // args.localSystem);
 
   crossSystem = if crossSystem0 == null then localSystem
                 else lib.systems.elaborate crossSystem0;
 
+  # Massage e into a NixOS module.
+  # We need all this stuff only because we want to support unknown options,
+  # without them this can be simplified a lot.
```

Comment:
I think this should also indicate how unknown options are not something to stay, which can justify this hacky implementation.